src/KustoExpressionParser.ts

```diff
@@ -399,17 +399,23 @@ const defaultTimeColumn = (columns?: AdxColumnSchema[], expression?: QueryExpres
   return toType(column.CslType, column.Name);
 };
 
-const isDynamic = (column: string): boolean => {
-  return !!(column && column.indexOf('[') > -1) || !!(column && column.indexOf('todynamic') > -1);
+const isDynamic = (column: string | AdxColumnSchema | undefined): boolean => {
+  if (column && typeof column === 'object') {
+    return column.Type === 'dynamic';
+  } else if (typeof column === 'string') {
+    return !!(column && column.indexOf('[') > -1) || !!(column && column.indexOf('todynamic') > -1);
```

Comment:
we probably don't need this case anymore, right? And we can simplify the column type to be only `AdxColumnSchema | undefined`